pkg/maps/auth/auth_map.go

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package auth
+
+import (
+	"fmt"
+	"sync"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/ebpf"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/policy"
+)
+
+const (
+	MapName    = "cilium_auth_map"
+	MaxEntries = 512000
```

Comment:
I think the max size for most eBPF maps is configurable using dedicated options for the agent. Would it make sense to have the same here?

Please update `Documentation/network/ebpf/maps.rst`.

Reply:
@qmonnet thanks for the valuable input.

the max size of the auth map is now configurable and documented!